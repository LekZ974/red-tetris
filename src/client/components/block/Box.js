import styled from 'react-emotion'
import {
  space,
  width,
  fontSize,
  textColor,
  bgColor,
  color,
  // typography
  fontFamily,
  textAlign,
  lineHeight,
  fontWeight,
  letterSpacing,
  // layout
  display,
  maxWidth,
  minWidth,
  height,
  maxHeight,
  minHeight,
  size,
  ratio,
  // flexbox
  alignItems,
  alignContent,
  justifyContent,
  flexWrap,
  flexDirection,
  flex,
  flexBasis,
  justifySelf,
  alignSelf,
  order,
  // grid
  gridGap,
  gridColumnGap,
  gridRowGap,
  gridColumn,
  gridRow,
  gridAutoFlow,
  gridAutoColumns,
  gridAutoRows,
  gridTemplateColumns,
  gridTemplateRows,
  // borders
  border,
  borderTop,
  borderRight,
  borderBottom,
  borderLeft,
  borders,
  borderColor,
  borderRadius,
  // misc
  boxShadow,
  background,
  backgroundImage,
  backgroundSize,
  backgroundPosition,
  backgroundRepeat,
  // position
  position,
  zIndex,
  top,
  right,
  bottom,
  left,
  // psuedoclasses
  hover,
  focus,
  active,
  disabled,
  // complex
  textStyle,
  colorStyle,
  buttonStyle
} from 'styled-system'

const Box = styled('div')(
  props => ({
    ...(props.flex === true
      ? {
        flex: 1,
        display: 'flex'
      }
      : {}),
    ...(props.center
      ? {
        display: 'flex',
        alignItems: 'center',
        justifyContent: 'center'
      }
      : {}),
    ...(props.container
      ? {
        display: 'flex',
        backgroundColor: '#F6F7FB',
        flex: 1
      }
      : {})
  }),
  space,
  width,
  fontSize,
  textColor,
  bgColor,
  color,
  // typography
  fontFamily,
  textAlign,
  lineHeight,
  fontWeight,
  letterSpacing,
  // layout
  display,
  maxWidth,
  minWidth,
  height,
  maxHeight,
  minHeight,
  size,
  ratio,
  // flexbox
  alignItems,
  alignContent,
  justifyContent,
  flexWrap,
  flexDirection,
  flex,
  flexBasis,
  justifySelf,
  alignSelf,
  order,
  // grid
  gridGap,
  gridColumnGap,
  gridRowGap,
  gridColumn,
  gridRow,
  gridAutoFlow,
  gridAutoColumns,
  gridAutoRows,
  gridTemplateColumns,
  gridTemplateRows,
  // borders
  border,
  borderTop,
  borderRight,
  borderBottom,
  borderLeft,
  borders,
  borderColor,
  borderRadius,
  // misc
  boxShadow,
  background,
  backgroundImage,
  backgroundSize,
  backgroundPosition,
  backgroundRepeat,
  // position
  position,
  zIndex,
  top,
  right,
  bottom,
  left,
  // psuedoclasses
  hover,
  focus,
  active,
  disabled,
  // complex
  textStyle,
  colorStyle,
  buttonStyle
)

Box.displayName = 'Box'

export default Box
