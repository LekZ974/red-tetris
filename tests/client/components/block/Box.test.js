import React from 'react';
import Box from '../../../../src/client/components/block/Box';
import renderer from 'react-test-renderer';

test('Box display', () => {
  const component = renderer.create(
    <Box>Hello</Box>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
