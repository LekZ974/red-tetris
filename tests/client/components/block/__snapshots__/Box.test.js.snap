// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box display 1`] = `
<div
  className="css-194we7b"
>
  Hello
</div>
`;
